projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -152,6 +153,56 @@ public static Memory<byte> GetHeartbeatFrame()
                 return new Memory<byte>(buffer, 0, FrameSize);
             }
         }
+
+        public static ReadOnlyMemory<byte> SerializeToFrames<T>(in T method, ushort channelNumber) where T : struct, IOutgoingCommand
+        {
+            var size = Method.FrameSize + method.GetRequiredBufferSize();
+            var array = ArrayPool<byte>.Shared.Rent(size);
+            var offset = Method.WriteTo(array, channelNumber, method);
+
+            if (offset != size)
+            {
+                throw new InvalidOperationException($"Serialized to wrong size, expect {size}, offset {offset}");
+            }
+
+            return new ReadOnlyMemory<byte>(array, 0, size);
```

Comment:
use .AsMemory(0, size) extension method for simplicity?

Reply:
AsMemory creates a Memory<byte> not a ReadOnlyMemory<byte> => implicit cast